src/app/account/keys/generate-keys.tsx

```diff
@@ -28,91 +47,111 @@ export const GenerateKeys: FC = () => {
-                        <Stack gap="xl">
-                            <Title>Private key</Title>
-                            <ScrollArea>{keys.privateKey}</ScrollArea>
-
+                        <Group>
                             <Group>
                                 <CopyButton value={keys.privateKey}>
                                     {({ copied, copy }) => (
                                         <Button
                                             disabled={copied}
                                             onClick={() => {
                                                 copy()
+                                                setIsKeyCopied(true)
                                             }}
                                         >
-                                            {copied ? 'Copied private key!' : 'Copy private key'}
+                                            Copy key
                                         </Button>
                                     )}
                                 </CopyButton>
                             </Group>
-                            <Text>
-                                This Private Key is unique to you. Please store your private key securely, such as in
-                                your password manager.{' '}
-                                <Text fw="bold" component="span">
-                                    It is important to note that SafeInsights does not store your private key.
-                                </Text>{' '}
-                                If lost, you cannot access encrypted results and will need to generate a new Private
-                                Key.
-                            </Text>
-                        </Stack>
-
-                        <Group>
-                            <Button onClick={open}>Go To Dashboard</Button>
+                            <Group>
+                                <Button variant="outline" onClick={() => openConfirmKeyCopied()}>
+                                    Go to dashboard
+                                </Button>
+                            </Group>
                         </Group>
+                        {isKeyCopied && (
+                            <Flex gap="xs">
+                                <Check size={16} color={theme.colors.green[9]} />
+                                <Text c="green.9" size="xs" fw={500}>
+                                    Copied!
+                                </Text>
+                            </Flex>
+                        )}
                     </Stack>
                 </Paper>
+                <ConfirmationModal onClose={closeConfirmKeyCopied} isOpen={confirmationOpened} keys={keys} />
             </Container>
         )
     }
+}
+
+const ConfirmationModal: FC<{ onClose: () => void; isOpen: boolean; keys: Keys }> = ({ onClose, isOpen, keys }) => {
+    const router = useRouter()
+
+    const { mutate: saveReviewerKey } = useMutation({
```

Comment:
capture pending for use on button below:
```suggestion
    const { mutate: saveReviewerKey, isPending: isSavingKey } = useMutation({
```